### PR TITLE
SI-6287 fixes synthetic symbol clashes in toolbox

### PR DESCRIPTION
**Pull request by xeno-by**
Apparently synthetic classes like $anonfun$1 have two properties:
1) Their names are generated using a counter unique to a compilation unit
2) After flatten they levitate to the nearest enclosing package

As a result if we use an empty package to wrap toolbox codegen,
then this package will soon be overflown by $anonfun$1 symbols, because:
1) New codegen session = new compilation unit = new counter which starts at 0
2) New codegen session = new anon funs that end up as children of empty package

Creating a freshly named package for each codegen session fixed the problem.
Now anonfuns from different sessions end up with different parents.

### CODE REVIEW

**Comment by xeno-by** [comment]
review by @paulp @retronym @gkossakowski

File: test/files/run/t5770.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import scala.reflect.runtime.{currentMirror => cm}
+import scala.tools.reflect._
+
+object Test extends App {
+  var i = 0
+  val action = reify { i += 1; println(i) }.tree
+
+  val tb1 = cm.mkToolBox()
+  tb1.eval(action) // works
+  tb1.eval(action) // error
```
Comment (gkossakowski) [inline test/files/run/t5770.scala]: Why here it's error and after `reset()` it's fine to call `eval(action)` twice?

Comment (xeno-by) [inline test/files/run/t5770.scala]: The test is just copy/pasted verbatim from the bug report. No idea what was the original cause of the problem - this bug has fixed itself over a few months.

Comment (gkossakowski) [inline test/files/run/t5770.scala]: I see. Can you clean up those comments? Comments in test-cases are supposed to explain what's going on and what's expected. This is not the case.

Comment (xeno-by) [inline test/files/run/t5770.scala]: Will do

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
LGTM.

**Pull request merged**